third_party/nvfuser/test/test_gpu3.cpp

```diff
@@ -7750,6 +7750,136 @@ TEST_F(NVFuserTest, FusionPredicateReductionInitGlobal_CUDA) {
       fe.kernel(), cg_outputs, inputs, {ref_t1, ref_t3}, __LINE__, __FILE__);
 }
 
+// Make sure invalid usage of index type is detected
+TEST_F(NVFuserTest, FusionCompileIndexType_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeSymbolicTensor(1, DataType::Half);
+  fusion.addInput(tv0);
+
+  auto tv1 = makeSymbolicTensor(1, DataType::Half);
+  fusion.addInput(tv1);
+
+  auto tv2 = add(tv0, tv1);
+  fusion.addOutput(tv2);
+
+  tv2->split(0, 256);
+  tv2->split(0, 1024);
+
+  MaxRootDomainInfoSpanningTree tree(tv2);
+  TransformPropagator tp(tv2);
+  tree.traverse(&tp);
+
+  inlineMost();
+
+  tv2->axis(1)->parallelize(ParallelType::BIDx);
+  tv2->axis(2)->parallelize(ParallelType::TIDx);
+
+  auto options = at::TensorOptions().dtype(at::kHalf).device(at::kCUDA, 0);
+  at::manual_seed(0);
+  at::Tensor t0 = at::randn({999}, options);
+  std::vector<c10::IValue> small_inputs = {t0, t0};
+
+  at::Tensor t0_large = at::randn({std::numeric_limits<int>::max()}, options);
```

Comment:
Would it be safer to use `(int64_t)std::numeric_limits<int>::max() + 1`? In theory `std::numeric_limits<int>::max()` is still indexable with 32bit indexing.

Reply:
This assumes how we determine the index mode, i.e., we only allow up to half of the int32 max. If this assumption breaks, the `TORCH_ERROR` about `large_inputs` should break, so this should be fine.